dojo-dungeon/src/starter/DojoRoom.java

```diff
@@ -0,0 +1,73 @@
+package starter;
+
+import contrib.level.generator.graphBased.levelGraph.Direction;
+import contrib.level.generator.graphBased.levelGraph.LevelGraph;
+import contrib.level.generator.graphBased.levelGraph.LevelNode;
+import core.Entity;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A room of the dojo-dungeon game. This class holds one level node and its entities.
+ *
+ * <p>This class is useful to build a level graph and to add later entities to it.
+ */
+public class DojoRoom {
```

Comment:
@tgrothe für mich sieht das wie eine art wrapper um LevelNode aus oder täusche ich mich da?

Reply:
> @tgrothe für mich sieht das wie eine art wrapper um LevelNode aus oder täusche ich mich da? 

Jap, das ist eine Art Wrapper. LevelNode ist an für sich final, und das wollte ich nicht aufbohren, weil das(?) Dojo doch darauf aufbauen soll, das heißt, die Basis möglichst nicht verändern soll.

Dann bleibt aber aus meiner Sicht nur eine Art Wrapper oder Kapselung übrig. Erweitern/ableiten geht ja nicht, wegen final.

Reply:
Verstehe. Da der DojoRoom (noch) nichts wirklich tut außer Getter/Setter: Den LevelNode einfach nur nutzen ist keine Option?

Ich frage nur, weil es schon eine große Anzahl teilweise verwirrender Klassen gibt. Wenn ihr natürlich den DojoRoom noch deutlich erweitern wollt, dann passt das.

Reply:
> Den LevelNode einfach nur nutzen ist keine Option?

Nein, wäre keine Option, weil wir zu jeden Raum genau ein Set mit Entities brauchen, das beim Anlegen der Klasse vorhanden sein muss. Dafür sind Klassen schließlich gedacht. ;)

Zudem könnte die open the doors-Methode vielleicht auch in die Raum-Klasse.

Reply:
> > Den LevelNode einfach nur nutzen ist keine Option?
> 
> Nein, wäre keine Option, weil wir zu jeden Raum genau ein Set mit Entities brauchen, das beim Anlegen der Klasse vorhanden sein muss. 

Ist das nicht bei LevelNode exakt auch so? Irgendwas verstehe ich nicht, glaube ich.

Reply:
> > > Den LevelNode einfach nur nutzen ist keine Option?
> > 
> > 
> > Nein, wäre keine Option, weil wir zu jeden Raum genau ein Set mit Entities brauchen, das beim Anlegen der Klasse vorhanden sein muss.
> 
> Ist das nicht bei LevelNode exakt auch so? Irgendwas verstehe ich nicht, glaube ich.

LevelNode setzt voraus, dass alle entities bereits beim Erstellen von LevelNode (Ctor) hinzugefügt werden und bekannt sind. Wir kennen die entities für einen Raum aber erst, wenn die einzelnen Räume generiert werden. Das ist das "Dilemma"...

Reply:
> LevelNode ist an für sich final, und das wollte ich nicht aufbohren, weil das(?) Dojo doch darauf aufbauen soll, das heißt, die Basis möglichst nicht verändern soll.

Prinzipiell finde ich das richtig - das Dojo soll möglichst auf dem Dungeon aufbauen, d.h. diesen nicht anfassen. Aber ich würde jetzt kein Dogma draus machen: Wenn Du eine gute Begründung hast, warum irgendwas angefasst werden soll, dann sollte man entsprechend drüber nachdenken, bevor man riesige Ersatzkonstrukte errichtet.

Hier in diesem konkreten Fall werde ich das Gefühl nicht los, dass Du eigentlich nur ein `LevelNode` brauchst. Der Wrapper `DojoRoom` geht ja eigentlich nicht über die Funktionalität von `LevelNode` hinaus - aber ich weiss natürlich nicht, wo Du noch hinwillst damit. Aber dann ist eine Ableitung der Klasse vielleicht einfacher und erspart zumindest diese ganzen Getter/Setter shenanigans, und dann wäre ich auch damit okay, das `final` in `LevelNode` rauszunehmen. Die Auswirkungen sind in diesem Fall ja zudem recht überschaubar, d.h. bestehender Code sollte dadurch nicht beeinflusst werden.

Reply:
Ok, dann ... -`final` bei `LevelNode` und +`extends LevelNode` bei `DojoRoom` ...

Dadurch entfällt das zusätzliche Halten von der `Set` für die entities, und der Code wird einfacher.

Reply:
fertig, sollte aber noch im ersten Kommentar erwähnt werden

Reply:
erster PR Kommentar ist aktualisiert